### PR TITLE
Fix expiration read another device

### PR DESCRIPTION
**Pull request by Bilb**
* auto delete unread DaR messages after 14 days
* add some messageInfo details hidden by default
* make the ctx menu expireTimer force refresh to the second (even if the mouse is not moving for some time)
* make the Set button on disappearing message sticky with a gradient so it is visible even if the window is too small

### CODE REVIEW

File: ts/components/conversation/message/message-content/MessageReactBar.tsx [hunk 1]
```diff
@@ -91,8 +94,7 @@ function useIsRenderedExpiresInItem(messageId: string) {
   return expiryDetails.expirationTimestamp;
 }
 
-function formatExpiry({ expirationTimestamp }: { expirationTimestamp: number }) {
-  const diffMs = expirationTimestamp - Date.now();
+function formatExpiry({ diffMs }: { diffMs: number }) {
```
Comment (yougotwill) [inline ts/components/conversation/message/message-content/MessageReactBar.tsx:97]: Not sure what diff means? Maybe rename it?

File: ts/components/conversation/message/message-content/MessageReactBar.tsx [hunk 2]
```diff
@@ -132,15 +134,25 @@ function formatExpiry({ expirationTimestamp }: { expirationTimestamp: number })
...
-  if (!expirationTimestamp) {
+  // this boolean is just used to forceRefresh the state when we get to display seconds in the contextmenu
+  const [refresh, setRefresh] = useBoolean(false);
+  const diffMs = (expirationTimestamp || 0) - Date.now();
+
+  useInterval(
+    () => {
+      setRefresh(!refresh);
+    },
+    diffMs > 0 && diffMs <= 2 * DURATION.MINUTES ? 500 : null
```
Comment (yougotwill) [inline ts/components/conversation/message/message-content/MessageReactBar.tsx:145]: 🧙

File: ts/components/conversation/right-panel/overlay/disappearing-messages/OverlayDisappearingMessages.tsx [hunk 3]
```diff
@@ -18,17 +18,22 @@ import {
...
 import { SessionButton } from '../../../../basic/SessionButton';
-import { SpacerLG, SpacerXL } from '../../../../basic/Text';
+import { SpacerLG } from '../../../../basic/Text';
 import { Header, HeaderSubtitle, HeaderTitle, StyledScrollContainer } from '../components';
 import { DisappearingModes } from './DisappearingModes';
 import { TimeOptions } from './TimeOptions';
 
-const StyledContainer = styled(Flex)`
+const StyledButtonContainer = styled.div`
+  background: linear-gradient(0deg, black, transparent);
```
Comment (yougotwill) [inline ts/components/conversation/right-panel/overlay/disappearing-messages/OverlayDisappearingMessages.tsx:27]: I'll allow it.

File: ts/components/conversation/right-panel/overlay/message-info/components/MessageInfo.tsx [hunk 4]
```diff
@@ -57,6 +64,48 @@ const showDebugLog = () => {
   ipcRenderer.send('show-debug-log');
 };
 
+const showDebugMessageInfo = false;
+
+const DebugMessageInfo = ({ messageId }: { messageId: string }) => {
```
Comment (yougotwill) [inline ts/components/conversation/right-panel/overlay/message-info/components/MessageInfo.tsx:69]: Should this not be in a separate component file and then imported?

File: ts/components/conversation/right-panel/overlay/message-info/components/MessageInfo.tsx [hunk 5]
```diff
@@ -57,6 +64,48 @@ const showDebugLog = () => {
   ipcRenderer.send('show-debug-log');
 };
 
+const showDebugMessageInfo = false;
```
Comment (yougotwill) [inline ts/components/conversation/right-panel/overlay/message-info/components/MessageInfo.tsx:67]: How do I use this? Is it in the `window` object?

File: ts/session/disappearing_messages/index.ts [hunk 6]
```diff
@@ -543,18 +544,40 @@ function getMessageReadyToDisappear(
...
-        1000}s ago, so with ${(expires_at - Date.now()) / 1000}s left`
-    );
-    messageModel.set({
-      expirationStartTimestamp,
-      expires_at,
-    });
+    /**
+     * Edge case: when we send a message before we poll for a message sent earlier, our convo volatile update will
+     * mark that incoming message as read right away (because it was sent earlier than our latest convolatile lastRead).
+     * To take care of this case, we need to check if the expiration of an incoming DaR, alreadt marked as read message looks to not have been updated yet.
```
Comment (yougotwill) [inline ts/session/disappearing_messages/index.ts:550]: `To take care of this case, we need to check if the expiration of an incoming DaR, alreadt marked as read message looks to not have been updated yet.` -> `To take care of this case, we need to check if the expiration of an incoming DaR is already marked as read and looks like it has not been updated yet.`

File: ts/session/disappearing_messages/index.ts [hunk 7]
```diff
@@ -543,18 +544,40 @@ function getMessageReadyToDisappear(
...
+     * The way we do it, is by checking that the swarm expiration is before (now + expireTimer).
+     * If it looks like this expiration was not updated yet, we need to trigger a UpdateExpiryJob for that message.
```
Comment (yougotwill) [inline ts/session/disappearing_messages/index.ts:552]: Noice

File: ts/session/disappearing_messages/index.ts [hunk 8]
```diff
@@ -543,18 +544,40 @@ function getMessageReadyToDisappear(
...
+      });
+    } else {
+      window.log.debug(
+        `incoming DaR message already read by another device but swarmExpiration seems NOT updated, forcing readAt NOW and triggering UpdateExpiryJob with ${expireTimer}s left`
+      );
+      messageModel.set({
+        expirationStartTimestamp: now,
+        expires_at: expirationNowPlusTimer,
+      });
+      // Ideally we would batch call those UpdateExpiry, but we can't currently and disappear v2 is already too complex as it is.
```
Comment (yougotwill) [inline ts/session/disappearing_messages/index.ts:578]: I don't see this edge case causing problems at all 🙃😂

File: ts/models/messageFactory.ts [hunk 9]
```diff
@@ -117,6 +117,7 @@ export function markAttributesAsReadIfNeeded(messageAttributes: MessageAttribute
       latestUnreadForThisConvo?.lastRead &&
       sentAt <= latestUnreadForThisConvo.lastRead
     ) {
+      // That message was sent before our last read timestamp for that conversation.
```
Comment (yougotwill) [inline ts/models/messageFactory.ts:120]: * The message that was sent

**Review by yougotwill — CHANGES_REQUESTED** [review]

File: ts/components/conversation/right-panel/overlay/disappearing-messages/OverlayDisappearingMessages.tsx (hunk 3 above)
Comment (Bilb) [reply ts/components/conversation/right-panel/overlay/disappearing-messages/OverlayDisappearingMessages.tsx:27]: bruh

File: ts/components/conversation/right-panel/overlay/message-info/components/MessageInfo.tsx (hunk 4 above)
Comment (Bilb) [reply ts/components/conversation/right-panel/overlay/message-info/components/MessageInfo.tsx:69]: I'd rather keep it here. This fine is small enough and I don't want that component be exported in the global namespace for no reason

File: ts/components/conversation/right-panel/overlay/message-info/components/MessageInfo.tsx (hunk 5 above)
Comment (Bilb) [reply ts/components/conversation/right-panel/overlay/message-info/components/MessageInfo.tsx:67]: set it to true? but don't commit it!

File: ts/components/conversation/message/message-content/MessageReactBar.tsx (hunk 2 above)
Comment (Bilb) [reply ts/components/conversation/message/message-content/MessageReactBar.tsx:145]: explained a bit my magic

File: ts/components/conversation/message/message-content/MessageReactBar.tsx (hunk 1 above)
Comment (Bilb) [reply ts/components/conversation/message/message-content/MessageReactBar.tsx:97]: renamed to formatTimeLeft

Comment (Bilb) [reply ts/components/conversation/message/message-content/MessageReactBar.tsx:97]: and renamed the variable itself

File: ts/components/conversation/right-panel/overlay/message-info/components/MessageInfo.tsx (hunk 5 above)
Comment (Bilb) [reply ts/components/conversation/right-panel/overlay/message-info/components/MessageInfo.tsx:67]: ok I'll make this more dynamic

**Review by yougotwill — APPROVED** [review]
LGTM 🚢

**Pull request merged**